crates/common/trie/node.rs

```diff
@@ -179,7 +179,7 @@ impl Node {
 fn decode_child(rlp: &[u8]) -> NodeHash {
     match decode_bytes(rlp) {
         Ok((hash, &[])) if hash.len() == 32 => NodeHash::Hashed(H256::from_slice(hash)),
-        Ok((&[], &[])) => NodeHash::Inline(vec![]),
-        _ => NodeHash::Inline(rlp.to_vec()),
+        Ok((&[], &[])) => NodeHash::default(),
+        _ => NodeHash::inline_from_slice(rlp),
```

Comment:
Maybe this match should be inside NodeHash module as well? I don't like the fact that inline_from_slice can panic and it's public. We should at least have that limited to a single module.

Reply:
I don't think the match should be go there, but i changed it a bit